torch/_vmap_internals.py

```diff
@@ -62,6 +108,30 @@ def _as_tuple(value, num_elements, error_message_lambda):
         raise ValueError(error_message_lambda())
     return value
 
+# Creates BatchedTensors for every Tensor in arg that should be batched.
+# Returns the (potentially) batched arguments and the batch_size.
+def _create_batched_inputs(in_dims, args, vmap_level, fn_name):
+    if not isinstance(in_dims, int) and not isinstance(in_dims, tuple):
+        raise ValueError(EXPECTED_IN_DIMS_TO_BE_INT_OR_TUPLE.format(
+            fn=fn_name, in_dims=in_dims, actual_type=str(type(in_dims))))
+
+    # NB: Checks that len(in_dims) == len(args) (if in_dims is a tuple).
+    in_dims_as_tuple = _as_tuple(
+        in_dims, len(args),
+        lambda: IN_DIMS_AND_NUM_INPUTS_MISMATCH.format(
+            fn=fn_name, in_dims=in_dims, num_inputs=len(args)))
+
+    if len(args) == 0:
+        raise ValueError(NO_INPUTS.format(fn=fn_name))
```

Comment:
Is there any reason to block the zero length args case, besides "then vmap doesn't do anything"? I'm thinking of how people have found it useful to do zero-size batches; it may be harmless to have zero length args (unless it is not?)

Reply:
The only two reasons I have are: (1) "then vmap doesn't do anything" and (2) "jax doesn't allow it". I agree that it seems harmless to have zero-length arguments.

It's not too hard to modify this to work so I'll add this as a follow-up for later (and think more about if it is actually harmless or not)